app/controllers/api/v1/visualization/clusters_controller.rb

```diff
@@ -102,9 +102,10 @@ def show
               render json: {error: "No cluster named #{cluster_name} could be found"}, status: 404 and return
             end
           end
-          viz_data = self.class.get_cluster_viz_data(@study, cluster, params)
-          if viz_data.nil?
-            render json: {error: 'Annotation could not be found'}, status: 404 and return
+          begin
+            viz_data = self.class.get_cluster_viz_data(@study, cluster, params)
+          rescue ArgumentError => e
+            render json: {error: e.message}, status: 422 and return
```

Comment:
(puts on HTTP status code hat) `422` is `Unprocessable Entity`, and deals specifically the case when the request had the correct syntax/instructions, but the server could not process them.  I've really only ever seen it applied to a `POST` or `PUT` when saving data, and the parameters supplied cause a validation error, in which case the server responds `422` because the request could not be completed as outlined.

In this particular case, we're trying to find some visualization data via a `GET`, but `ClustersController.get_cluster_viz_data` doesn't find what we asked for.  I would posit then that the correct response code would be `404` or `400`, though it would depend on what the request parameters actually were.  If we were missing a required parameter, `400` would be more correct, but if the sum of all params equates to an empty response, then `404`.